ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,282 @@
+import logging
+import time
+import base64
+import os
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import run_cmd
+from tests import helpers
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.framework import config
+
+
+logger = logging.getLogger(__name__)
+
+
+def change_registry_backend_to_ocs():
+    """
+    Function to deploy registry with OCS backend.
+
+    Raises:
+        AssertionError: When failure in change of registry backend to OCS
+    """
+    pv_obj = helpers.create_pvc(
+        sc_name=constants.DEFAULT_SC_CEPHFS, pvc_name='registry-cephfs-rwx-pvc',
+        namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE, size='100Gi',
+        access_mode=constants.ACCESS_MODE_RWX
+    )
+    helpers.wait_for_resource_state(pv_obj, 'Bound')
+    ocp_obj = ocp.OCP(
+        kind=constants.CONFIG, namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+    )
+    param_cmd = f'[{{"op": "add", "path": "/spec/storage", "value": {{"pvc": {{"claim": "{pv_obj.name}"}}}}}}]'
+    assert ocp_obj.patch(
+        resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME, params=param_cmd
+    ), f"Registry pod storage backend to OCS is not success"
+
+    # Validate registry pod status
+    validate_registry_pod_status()
+
+    # Validate pvc mount in the registry pod
+    validate_pvc_mount_on_registry_pod()
+
+
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+
+    Returns:
+        pod_obj (obj): Registry pod obj
+
+    Raises:
+        UnexpectedBehaviour: When image-registry pod is not present.
+    """
+    # Sometimes when there is a update in config crd, there will be 2 registry pods
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    iteration = 0
+    while True:
+        iteration += 1
+        pod_data = pod.get_pods_having_label(
+            label='docker-registry=default', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+        )
+        pod_obj = [pod.Pod(**data) for data in pod_data]
```

Comment:
pod object looks like the list to me here and you are saying in the doc that you are returning 
pod_obj (obj): Registry pod obj
But on the line 75 seems you are returning this list of pod objects. Isn't that suppose to return pod_obj[0] then? Or did I miss something?

Reply:
While changing the registry backend for some seconds or minutes there will be 2 pods running i.e. Existing one which will be in terminating and new pod which will be Up. Expectation is only one pod obj but during the backend change process there will be 2 pods.

Reply:
ack, updated the Returns comment for line 75